mint-client/src/clients/user.rs

```diff
@@ -323,6 +326,65 @@ impl UserClient {
             .await
             .map_err(|_| ClientError::WaitContractTimeout)?
     }
+
+    pub async fn create_invoice_and_offer<R: RngCore + CryptoRng>(
+        &self,
+        amount: Amount,
+        mut rng: R,
+    ) -> Result<(KeyPair, Invoice), ClientError> {
+        let (payment_keypair, payment_public_key) =
+            self.context.secp.generate_schnorrsig_keypair(&mut rng);
+        let raw_payment_secret = payment_public_key.serialize();
+        let payment_hash = bitcoin::secp256k1::hashes::sha256::Hash::hash(&raw_payment_secret);
+        let payment_secret = PaymentSecret(raw_payment_secret);
+
+        // Temporary lightning node pubkey
+        let (node_secret_key, node_public_key) = self.context.secp.generate_keypair(&mut rng);
+
+        let invoice = InvoiceBuilder::new(Currency::Regtest)
```

Comment:
Could get the network from `self.context.config.wallet.network`

Reply:
Added a `network_to_currency` helper function in this file to do the conversion. Can't implement in `lightning-invoice` because rust-bitcoin isn't a dependency!